odo/convert.py

```diff
@@ -213,14 +213,24 @@ def _():
             yield convert(pd.DataFrame, [], **kwargs)
     else:
         df = convert(pd.DataFrame, first, **kwargs)
+        df1, n1 = _add_index(df, 0)
 
         def _():
-            yield df
+            n = n1
+            yield df1
             for i in rest:
-                yield convert(pd.DataFrame, i, **kwargs)
+                df = convert(pd.DataFrame, i, **kwargs)
+                df, n = _add_index(df, n)
+                yield df
     return chunks(pd.DataFrame)(_)
 
 
+def _add_index(df, start):
+    idx = pd.Index(np.arange(start, start + len(df)))
```

Comment:
I feel like we should use a `RangeIndex` here for newer versions of pandas. We can conditionally define some `mkindex` function and maybe store it as a default arg to this `add_index` so that we don't pay a perf penalty for this.